src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -295,6 +328,94 @@ void ScheduleTask(DateTimeOffset utcNow)
             }
         }
 
+        private (long, long) Sum()
+        {
+            lock (_allStats!)
+            {
+                long hits = _accumulatedStats!.Hits;
+                long misses = _accumulatedStats.Misses;
+
+                foreach (WeakReference<Stats> wr in _allStats)
+                {
+                    if (wr.TryGetTarget(out Stats? stats))
+                    {
+                        hits += Interlocked.Read(ref stats.Hits);
+                        misses += Interlocked.Read(ref stats.Misses);
```

Comment:
Technically, these only need to be interlocked on 32-bit.  In practice, the overhead is small enough and this called infrequently enough that it probably doesn't matter.

Reply:
These only need to use `Volatile.Read` (which will be emitted as a regular read on 64-bit platforms).